### PR TITLE
Milestone 6 friendship v2

### PR DESCRIPTION
**Pull request by eapenzacharias**
- Added mutual friends to views
- Fixed user profile views
- implemented inverse friendships

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -2,4 +2,36 @@
...
+        <li>
+          <p><strong><%= "#{x.name}" %> </strong><%="#{x.email}" %> </p>
+          <p><%= link_to 'Profile', user_path(x.id) %></p>
+        </li>
+      <% end %>
+    </ul>
+  </div>
+<div class='col-md-5'>
+    <h3>Mutual Friends</h3>
+    <%@mutual_friendships = Friendship.where(user_id: current_user, confirmed: true, friend_id: (Friendship.where(user_id: @users.id, confirmed: true).select('friend_id'))) %>
```
Comment (samgaco) [inline app/views/users/show.html.erb]: - [ ] Try to keep the queries out of the views, you can define this query in the Friendship model (same for all the other queries in your views.)

Comment (eapenzacharias) [reply app/views/users/show.html.erb]: We have removed the query from the view and placed it in the controller and fixed the error on the user page. We will work on the CSS after finishing with the Milestone 7

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+  <h3><%= @user.email %></h3>
+  <p><%= @user.created_at %></p>
+  <% if (!current_user.friends.include? @user)&&(current_user != @user)%>
+    <%= link_to "Add Friend", friendships_path(:friend_id => @user), :method => :post %>
+  <% end %>
+</div>
+<div class='col-md-7'>
+    <h3><%= "#{@user.name.capitalize}'s" %> Friends</h3>
+    <ul>
+    <%@friendships = Friendship.where(user_id: @user.id, confirmed: true)%>
```
Comment (samgaco) [inline app/views/users/show.html.erb]: - [ ] Try to keep the queries out of the views

**Comment by eapenzacharias** [comment]
We are still having trouble preventing 
- the user from sending friend request to himself
- user A sends a request, user B can still send another request to user A

File: app/models/user.rb [hunk 3]
```diff
@@ -6,6 +6,7 @@ class User < ApplicationRecord
   has_many :friends, through: :friendships
   has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
   has_many :inverse_friends, through: :inverse_friendships, source: :user
+  has_many :confirmed_friendships, -> { where confirmed: true }, class_name: 'Friendship'
```
Comment (mosaaleb) [inline app/models/user.rb]: - [ ] There is a duplicate association for `inverse_friendships` here.Let's remove the first one.

Comment (lilleonz) [reply app/models/user.rb]: We are using it if we take it out then we have errors.

**Pull request merged**